system-parachains/asset-hubs/asset-hub-kusama/src/governance/origins.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot. If not, see <http://www.gnu.org/licenses/>.
+
+//! Custom origins for governance interventions.
+
+pub use pallet_custom_origins::*;
+
+// From https://github.com/polkadot-fellows/runtimes/blob/7bbf00566d86d51fcd5582779e7e9c37a814405e/relay/polkadot/src/governance/origins.rs#L21-L154
+#[frame_support::pallet]
+pub mod pallet_custom_origins {
+	use crate::{Balance, GRAND, QUID};
+	use frame_support::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[derive(
+		PartialEq,
+		Eq,
+		Clone,
+		MaxEncodedLen,
+		Encode,
+		Decode,
+		DecodeWithMemTracking,
+		TypeInfo,
+		RuntimeDebug,
+	)]
+	#[pallet::origin]
+	pub enum Origin {
+		/// Origin able to cancel slashes and manage minimum commission.
+		StakingAdmin,
+		/// Origin for spending up to $10,000,000 DOT from the treasury as well as generally
+		/// administering it.
+		Treasurer,
+		/// Origin for managing the composition of the fellowship.
+		FellowshipAdmin,
+		/// Origin for managing the registrar and permissioned HRMP channel operations.
+		GeneralAdmin,
+		/// Origin for starting auctions.
+		AuctionAdmin,
+		/// Origin able to force slot leases.
+		LeaseAdmin,
+		/// Origin able to cancel referenda.
+		ReferendumCanceller,
+		/// Origin able to kill referenda.
+		ReferendumKiller,
+		/// Origin able to spend around $250 from the treasury at once.
+		SmallTipper,
+		/// Origin able to spend around $1,000 from the treasury at once.
+		BigTipper,
+		/// Origin able to spend around $10,000 from the treasury at once.
+		SmallSpender,
+		/// Origin able to spend around $100,000 from the treasury at once.
+		MediumSpender,
+		/// Origin able to spend up to $1,000,000 DOT from the treasury at once.
+		BigSpender,
```

Comment:
Copied from AHP? So which one is correct here - $250 vs 1 KSM or $1,000,000 DOT vs 5,000 KSM?
```suggestion
		/// Origin able to spend up to 1 KSM from the treasury at once.
		SmallTipper,
		/// Origin able to spend up to 5 KSM from the treasury at once.
		BigTipper,
		/// Origin able to spend up to 50 KSM from the treasury at once.
		SmallSpender,
		/// Origin able to spend up to 500 KSM from the treasury at once.
		MediumSpender,
		/// Origin able to spend up to 5,000 KSM from the treasury at once.
		BigSpender,
```

Reply:
yes, its all copied. thank you. I still need review this myself.